python_modules/dagster/dagster/_core/definitions/events.py

```diff
@@ -149,6 +149,17 @@ def __eq__(self, other: object) -> bool:
             and self.tags == other.tags
         )
 
+    def with_metadata(self, metadata: Optional[Mapping[str, RawMetadataValue]]) -> "Output":
```

Comment:
We should separate this out into its own PR, since this is a core lib change.

Reply:
#21666